pkg/algorithm/cstorpoolselect/v1alpha1/select.go

```diff
@@ -103,6 +113,93 @@ type policy interface {
 	filter(*csp.CSPList) (*csp.CSPList, error)
 }
 
+// scheduleWithOverProvisioningAwareness is a pool
+// selection implementation.
+type scheduleWithOverProvisioningAwareness struct {
+	overProvisioning bool
+	spcName          string
+	volumeNamespace  string
+	capacity         resource.Quantity
+	err              []error
+}
+
+// priority returns the priority of the
+// policy implementation
+func (p scheduleWithOverProvisioningAwareness) priority() priority {
+	return highPriority
+}
+
+// name returns the name of the policy
+// implementation
+func (p scheduleWithOverProvisioningAwareness) name() policyName {
+	return overProvisioningPolicy
+}
+
+// filter selects the pools available on the host
+// for which the policy has been applied
+func (p scheduleWithOverProvisioningAwareness) filter(pools *csp.CSPList) (*csp.CSPList, error) {
+	if len(p.err) > 0 {
+		return nil, errors.Errorf("failed to fetch overprovisioning details:%v", p.err)
+	}
+
+	if p.overProvisioning {
+		klog.Info("Overprovisioning restriction policy not added as overprovisioning is enabled")
+		return pools, nil
+	}
+
+	filteredPools := csp.ListBuilder().List()
+	for _, pool := range pools.Items {
+		volCap, err := getAllVolumeCapacity(pool.Object)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get capacity consumed by existing volumes on pool %s ", pool.Object.UID)
```

Comment:
if an error occurs while getting the volume capacities of one CSP we are returning error without checking other pools. Can we log the error and continue to check with other pools? So that in a single replica case we will continue with other pools instead of throwing error.